### PR TITLE
Cadc 10630 Updated LoginServlet to prevent incorrect username in tokens

### PR DESCRIPTION
**Pull request by yeunga**
Added logic in LoginServlet to remove the principal created from a userID with different cases than the one in ldap.

### CODE REVIEW

File: cadc-access-control-server/src/main/java/ca/nrc/cadc/auth/AuthenticatorImpl.java [hunk 1]
```diff
@@ -147,6 +149,21 @@ public void augmentSubject(final Subject subject)
...
+                Principal princ = iterator.next();
+                if (princ instanceof HttpPrincipal) {
+                    toBeRemoved = princ;
+                    break;
+                }
+            }
+            if (toBeRemoved != null) {
+                subject.getPrincipals().remove(toBeRemoved);
+            }
+
```
Comment (brianmajor) [inline cadc-access-control-server/src/main/java/ca/nrc/cadc/auth/AuthenticatorImpl.java:166]: Like we discussed, I think all you need to do is, at this same place in the code:
`subject.getPrincipals().clear()`

**Review by brianmajor — CHANGES_REQUESTED** [review]

File: cadc-access-control-server/src/main/java/ca/nrc/cadc/auth/AuthenticatorImpl.java (hunk 1 above)
Comment (brianmajor) [reply cadc-access-control-server/src/main/java/ca/nrc/cadc/auth/AuthenticatorImpl.java:166]: Actually, only removing the HttpPrincipals would be safer.  So:
`subject.getPrincipals(HttpPrincipal.class).clear()`

Comment (yeunga) [reply cadc-access-control-server/src/main/java/ca/nrc/cadc/auth/AuthenticatorImpl.java:153]: Right. Thanks. Done.

**Review by brianmajor — DISMISSED** [review]
Looks good.

File: cadc-access-control-server/src/main/java/ca/nrc/cadc/auth/AuthenticatorImpl.java [hunk 2]
```diff
@@ -147,6 +149,18 @@ public void augmentSubject(final Subject subject)
             PluginFactory pluginFactory = new PluginFactory();
             UserPersistence userPersistence = pluginFactory.createUserPersistence();
             Principal ldapPrincipal = getLdapPrincipal(subject);
+
+            // Remove the HttpPrincipal in subject, if any
```
Comment (brianmajor) [inline cadc-access-control-server/src/main/java/ca/nrc/cadc/auth/AuthenticatorImpl.java:153]: Could you please add a comment as to why the http principal is being removed here?

File: cadc-access-control-server/src/main/java/ca/nrc/cadc/auth/AuthenticatorImpl.java [hunk 3]
```diff
@@ -147,6 +149,18 @@ public void augmentSubject(final Subject subject)
...
+            Set<Principal> toBeRemoved = new HashSet<Principal>();
+            subject.getPrincipals().forEach(p -> {
+                if (p instanceof HttpPrincipal) {
+                    toBeRemoved.add(p);
+                }
+            });
+            toBeRemoved.forEach(p -> {
+                subject.getPrincipals().remove(p);
+            });
+
```
Comment (brianmajor) [inline cadc-access-control-server/src/main/java/ca/nrc/cadc/auth/AuthenticatorImpl.java:163]: Would `removeAll(toBeRemoved)` be easier to read here?

File: cadc-access-control-server/src/main/java/ca/nrc/cadc/auth/AuthenticatorImpl.java [hunk 4]
```diff
@@ -147,6 +149,18 @@ public void augmentSubject(final Subject subject)
             PluginFactory pluginFactory = new PluginFactory();
             UserPersistence userPersistence = pluginFactory.createUserPersistence();
             Principal ldapPrincipal = getLdapPrincipal(subject);
+
+            // Remove the HttpPrincipal in subject, if any
+            Set<Principal> toBeRemoved = new HashSet<Principal>();
```
Comment (brianmajor) [inline cadc-access-control-server/src/main/java/ca/nrc/cadc/auth/AuthenticatorImpl.java:154]: Does this work?  `toBeRemoved = subject.getPrincipals(HttpPrincipal.class)`

**Review by brianmajor — CHANGES_REQUESTED** [review]

File: cadc-access-control-server/src/main/java/ca/nrc/cadc/auth/AuthenticatorImpl.java (hunk 3 above)
Comment (yeunga) [reply cadc-access-control-server/src/main/java/ca/nrc/cadc/auth/AuthenticatorImpl.java:162]: Good idea. This does work.

File: cadc-access-control-server/src/main/java/ca/nrc/cadc/auth/AuthenticatorImpl.java (hunk 4 above)
Comment (yeunga) [reply cadc-access-control-server/src/main/java/ca/nrc/cadc/auth/AuthenticatorImpl.java]: Great idea! It does. Now we are back to a one liner.

**Review by brianmajor — APPROVED** [review]

**Pull request merged**